### PR TITLE
Add integer setting values

### PR DESCRIPTION
**Pull request by alexdrean**
Add integer settings to handle multipliers for speed and fly.

Fly 1-10
Speed 1-5

Demo: https://youtu.be/GHS0TL1_DrE

This is by no means complete yet and there are still a couple fixme items I need to go through.
I would appreciate your feedback though.

Right now I am using a little hack to store values while settings are toggled off: negative values. I would like to get rid of this and add a proper value + status instead.

The migration from BOOLEAN to INTEGER should not create any problems on MySQL/MariaDB as BOOLEAN is an alias for INTEGER (or at least it is with InnoDB).

### CODE REVIEW

File: playersettings-plugin/src/main/java/me/limbo56/playersettings/settings/SimpleSettingWatcher.java [hunk 1]
```diff
@@ -24,24 +25,28 @@ public SimpleSettingWatcher(Player owner, Map<String, Setting> settingRegistry,
...
-    public void setValue(Setting setting, boolean value, boolean silent) {
+    public void setValue(Setting setting, int value, boolean silent) {
         if (!getOwner().hasPermission("settings." + setting.getRawName())) {
             PlayerUtils.sendConfigMessage(getOwner(), "settings.noPermission");
             return;
         }
 
         getStored().replace(setting, value);
-        Bukkit.getPluginManager().callEvent(new SettingUpdateEvent(getOwner(), setting, value));
+        Runnable runnable = () -> Bukkit.getPluginManager().callEvent(new SettingUpdateEvent(getOwner(), setting, value));
```
Comment (alexdrean) [inline playersettings-plugin/src/main/java/me/limbo56/playersettings/settings/SimpleSettingWatcher.java]: Fixes a crash when called from LoadPlayerTask

File: playersettings-plugin/src/main/java/me/limbo56/playersettings/settings/defined/FlySetting.java [hunk 2]
```diff
@@ -35,15 +35,24 @@ public int getSlot() {
...
     }
 
     public static class FlySettingCallback implements SettingCallback {
         @Override
-        public void notifyChange(Setting setting, Player player, boolean newValue) {
-            player.setAllowFlight(newValue);
-            player.setFlying(newValue);
+        public void notifyChange(Setting setting, Player player, int newValue) {
+            GameMode gm = player.getGameMode();
+            if (gm != GameMode.CREATIVE && gm != GameMode.SPECTATOR) {
```
Comment (alexdrean) [inline playersettings-plugin/src/main/java/me/limbo56/playersettings/settings/defined/FlySetting.java:52]: We don't really mean to remove creative players' ability to fly, do we?
This makes it act as a speed multiplier only when creative

File: pom.xml [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <revision>5.0.3</revision>
+        <revision>5.1.0</revision>
```
Comment (alexdrean) [inline pom.xml:14]: or whatever

**Comment by limbo56** [comment]
Hey @alexdrean, this is an awesome addition!

Your changes should be included once PR #32 is merged in. I have also added you as a contributor to the plugin's Spigot page. 

Thanks a lot for your help 🙌

**Pull request closed without merging**